lib/node_modules/@stdlib/blas/base/isamax/src/isamax_cblas.c

```diff
@@ -18,15 +18,37 @@
 
 #include "stdlib/blas/base/isamax.h"
 #include "stdlib/blas/base/isamax_cblas.h"
+#include "stdlib/blas/base/shared.h"
+#include "stdlib/strided/base/min_view_buffer_index.h"
 
 /**
 * Finds the index of the first element having the maximum absolute value.
 *
 * @param N        number of indexed elements
 * @param X        input array
-* @param strideX  stride length
+* @param strideX  X stride length
 * @return         index value
 */
-int c_isamax( const int N, const float *X, const int strideX ) {
-	return cblas_isamax( N, X, strideX );
+CBLAS_INT API_SUFFIX(c_isamax)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX ) {
+	if ( strideX < 0 ) {
+		return N - 1 - API_SUFFIX(cblas_isamax)( N, X, -strideX );
+	}
+	return API_SUFFIX(cblas_isamax)( N, X, strideX );
+}
+
+/**
+* Finds the index of the first element having the maximum absolute value using alternative indexing semantics.
+*
+* @param N        number of indexed elements
+* @param X        input array
+* @param strideX  X stride length
+* @param offsetX  starting index for X
+* @return         index value
+*/
+CBLAS_INT API_SUFFIX(c_isamax_ndarray)( const CBLAS_INT N, const float *X, const CBLAS_INT strideX, const CBLAS_INT offsetX ) {
+	X += stdlib_strided_min_view_buffer_index( N, strideX, offsetX ); // adjust array pointer
+	if ( strideX < 0 ) {
+		return N - 1 - API_SUFFIX(cblas_isamax)( N, X, -strideX );
```

Comment:
After looking at this, I don't think this actually works. It only works if the maximum absolute value occurs only once in an array. If there are two or more equivalent values, for a negative stride, what is returned will be incorrect. In short, flipping the stride has the effect of returning either the first or the last maximum absolute value.

I am not sure there is a workaround when calling into CBLAS.

Reply:
I think the only workaround is performing an explicit data copy which reverses an array having a negative stride before calling into CBLAS.

Reply:
@kgryte, I have applied the changes you can review it now if anything is incorrect.